pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -208,20 +231,116 @@ func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
 	return reconcile.Result{Requeue: true}, nil
 }
 
+// mergeValues merges a map containing input values on top of a map containing
+// base values, giving preference to the base values for conflicts
+func mergeValues(base map[string]interface{}, input map[string]interface{}) map[string]interface{} {
+	if base == nil {
+		base = make(map[string]interface{}, 1)
+	}
+
+	for key, value := range input {
+		//if the key doesn't already exist, add it
+		if _, exists := base[key]; !exists {
+			base[key] = value
+			continue
+		}
+
+		// at this point, key exists in both input and base.
+		// If both are maps, recurse.
+		// If only input is a map, ignore it. We don't want to overrwrite base.
+		// If both are values, again, ignore it since we don't want to overrwrite base.
+		if baseKeyAsMap, baseOK := base[key].(map[string]interface{}); baseOK {
+			if inputAsMap, inputOK := value.(map[string]interface{}); inputOK {
+				base[key] = mergeValues(baseKeyAsMap, inputAsMap)
+			}
+		}
+	}
+	return base
+}
+
+func (r *ControlPlaneReconciler) getSMCPTemplate(name string) (v1.ControlPlaneSpec, error) {
+	if strings.Contains(name, "/") {
+		return v1.ControlPlaneSpec{}, fmt.Errorf("template name contains invalid character '/'")
+	}
+
+	templateContent, err := ioutil.ReadFile(smcpTemplatePath + name)
+	if err != nil {
+		//if we can't read from the user template path, try from the default path
+		//we use two paths because Kubernetes will not auto-update volume mounted
+		//configmaps mounted in directories with pre-existing content
+		defaultTemplateContent, defaultErr := ioutil.ReadFile(smcpDefaultTemplatePath + name)
+		if defaultErr != nil {
+			return v1.ControlPlaneSpec{}, fmt.Errorf("template cannot be loaded from user or default directory. Error from user: %s. Error from default: %s", err, defaultErr)
+		}
+		templateContent = defaultTemplateContent
+	}
+
+	var template v1.ServiceMeshControlPlane
+	if err = yaml.Unmarshal(templateContent, &template); err != nil {
```

Comment:
Should we be using k8s decoder for this?  I believe that's the proper way to deserialize a k8s resource.

Reply:
This relies on the scheme to create the required resources, which is nil in the controlplane. Leaving as is for now. Created maistra-720 to look into this.

Reply:
This is nil, https://github.com/Maistra/istio-operator/blob/maistra-1.0/pkg/controller/servicemesh/controlplane/controller.go#L132?  Somehow the scheme is there, because the client uses it to deserialize the control plane object you're working on.  https://github.com/Maistra/istio-operator/blob/maistra-1.0/cmd/manager/main.go#L93